src/app/beer_garden/scheduler.py

```diff
@@ -647,7 +653,13 @@ def execute_job(job_id: str) -> Job:
     Returns:
         The spawned Request
     """
-    return db.query_unique(Job, id=job_id, raise_missing=True)
+    job = db.query_unique(Job, id=job_id, raise_missing=True)
+    job.reset_interval = reset_interval
+
+    if reset_interval and job.trigger_type != "interval":
+        raise ModelValidationError()
```

Comment:
As alluded to in another comment, you should have a message here with this exception.

```python
raise ModelValidationError("reset_interval can only be used with trigger type of interval")
```

Reply:
As a side note, `ModelValidationError` doesn't quite feel like the right exception here, but I don't feel super strongly about it.  Just a regular `ValueError` would probably have been my pick.  For reference, `ValueError` is described as:

> Raised when an operation or function receives an argument that has the right type but an inappropriate value.

That feels like it describes this situation.